packages/labs/analyzer/src/lib/model.ts

```diff
@@ -296,6 +323,78 @@ export class VariableDeclaration extends Declaration {
   }
 }
 
+export interface FunctionLikeInit extends NodeJSDocInfo {
+  name: string;
+  parameters?: Parameter[] | undefined;
+  return?: Return | undefined;
+}
+
+export class FunctionDeclaration extends Declaration {
+  parameters?: Parameter[] | undefined;
+  return?: Return | undefined;
+  constructor(init: FunctionLikeInit) {
+    super(init);
+    this.parameters = init.parameters;
+    this.return = init.return;
+  }
+}
+
+export type Privacy = 'public' | 'private' | 'protected';
+
+export interface SourceReference {
+  href: string;
+}
+
+export interface ClassMethodInit extends FunctionLikeInit {
+  static?: boolean | undefined;
+  privacy?: Privacy | undefined;
+  inheritedFrom?: Reference | undefined;
+  source?: SourceReference | undefined;
+}
+
+export class ClassMethod extends Declaration {
+  static?: boolean | undefined;
+  privacy?: Privacy | undefined;
+  inheritedFrom?: Reference | undefined;
+  source?: SourceReference | undefined;
+  parameters?: Parameter[] | undefined;
+  return?: Return | undefined;
+  constructor(init: ClassMethodInit) {
+    super(init);
+    this.static = init.static;
+    this.privacy = init.privacy;
```

Comment:
It's pretty important that this is correct and it shouldn't change from the base implementation so could/should this read from the `inheritedFrom` reference if it's not provided? (also for ClassField)

Reply:
Per discussion in chat, it's legal (though probably mostly unintentional) to widen the visibility, so not sure we should/can do anything safe here.

> or does privacy need to be brought forward from the base class?

I think the conclusion here is "yes", the author should carry the visibility forward when overriding. I will add this to the doc I'm writing re: API doc best practices.